src/main/scala/org/broadinstitute/dsde/workbench/sam/service/PolicyEvaluatorService.scala

```diff
@@ -35,6 +35,16 @@ class PolicyEvaluatorService(
   }
 
   def hasPermission(resource: FullyQualifiedResourceId, action: ResourceAction, userId: WorkbenchUserId): IO[Boolean] = {
+    // first attempt the cheap, short circuit and fallback to the full check if it returns false
+    for {
```

Comment:
nitpicky, but I don't think 'short circuit' is quite the right metaphor, and seeing 'short circuit' in an auth check method name is unsettling.  
maybe hasPermissionShallowCheck?

Reply:
yeah -- i looked for a better name.  I liked short circuit because the procedure here *is* a short-circuited boolean OR evaluation (e.g A || B returns after evaluating A if it is true and continues on to evaluate B if A was false).  The fact that A is shallow doesn't matter here.

However, I could see changing the name of the `hasPermissionShortCircuit` to `hasPermissionShallowCheck` or something since the short circuiting is actually above.